skipchain/skipchain.go

```diff
@@ -664,16 +888,104 @@ func (s *Service) newBlockEnd(sb *SkipBlock) bool {
 	return true
 }
 
+// authenticate searches if this node or any follower-node can verify the
+// schnorr-signature.
+func (s *Service) authenticate(msg []byte, sig []byte) bool {
+	if err := schnorr.Verify(Suite, s.ServerIdentity().Public, msg, sig); err == nil {
```

Comment:
At this point, msg and sig are untrusted input from the attacker. convince me that (1) the amount of mem/cpu they can consume here is bounded (I'm already 99% convinced), and (2) even though schnorr.Verify is not constant time, there is no timing channel that leaks private data.

Reply:
If you DOS the system, you can consume a lot of CPU here.
The data you leak is how many clients are signed up to the service.